Marlin/src/gcode/motion/G0_G1.cpp

```diff
@@ -68,6 +70,19 @@ void GcodeSuite::G0_G1(
 
     get_destination_from_command(); // For X Y Z E F
 
+    #if ENABLED(MAGNETIC_PARKING_EXTRUDER) && HAS_HOTEND_OFFSET
+      #if ENABLED(MPE_Z_OFFSET)
+        if (active_extruder)
+          destination[Z_AXIS] -= hotend_offset[Z_AXIS][active_extruder];
+      #endif
+      #if ENABLED(MPE_XY_OFFSET)
+        if (active_extruder) {
+          destination[X_AXIS] -= hotend_offset[X_AXIS][active_extruder];
+          destination[Y_AXIS] -= hotend_offset[Y_AXIS][active_extruder];
+        }
+      #endif
+    #endif
```

Comment:
This is a redundant and unnecessary hack. The `hotend_offset` should already have been applied to the `current_position` in `tool_change`, so that the current position already refers to the position of the selected hotend.

Reply:
Agree. 
I had an issue but it was now solved by pulling from upstream. 
I will revert this file to to head.

In future i will first discuss issues before try to solve them and i will make pull requests atomar . sorry for that.